lib/matplotlib/cm.py

```diff
@@ -177,23 +136,67 @@ def register(self, cmap, *, name=None, force=False):
             registered name. True supports overwriting registered colormaps
             other than the builtin colormaps.
         """
+        if not isinstance(cmap, colors.Colormap):
+            raise ValueError("You must pass a Colormap instance. "
+                             f"You passed {cmap} a {type(cmap)} object.")
+
         name = name or cmap.name
-        if name in self and not force:
-            raise ValueError(
-                f'A colormap named "{name}" is already registered.')
-        register_cmap(name, cmap.copy())
+        if name in self:
+            if not force and name in self._builtin_cmaps:
+                msg = f"Trying to re-register the builtin cmap {name!r}."
+                raise ValueError(msg)
+            elif not force:
+                raise ValueError(
+                    f'A colormap named "{name}" is already registered.')
+            else:
+                _api.warn_external(f"Trying to register the cmap {name!r} "
+                                   "which already exists.")
 
+        self._cmaps[name] = cmap.copy()
+
+    def unregister(self, name):
+        """
+        Remove a colormap from the registry.
+
+        You may not remove built-in colormaps.
+
+        If the named colormap is not registered, returns with no error, raises
+        if you try to de-register a default colormap.
+
+        .. warning::
+
+            The ColormapRegistry is a shared namespace that may be used
+            by multiple packages. Use `unregister` only if you know you
+            have registered that name before. In particular, do not
+            unregister just in case to clean the name before registering a
+            new colormap.
+
+        Parameters
+        ----------
+        name : str
+            The name of the colormap to be removed
```

Comment:
```suggestion
            The name of the colormap to be removed.
```
Following numpydoc conventions, also ellipses are terminated with a period.